multibranch/src/test/java/org/jenkinsci/plugins/workflow/multibranch/WorkflowMultiBranchProjectTest.java

```diff
@@ -116,4 +125,35 @@
         assertEquals(Collections.<Class<? extends BranchProperty>>emptySet(), clazzes);
     }
 
+    @SuppressWarnings("rawtypes")
+    @Test public void applicableSCMs() throws Exception {
+        final WorkflowMultiBranchProject mp = r.jenkins.createProject(WorkflowMultiBranchProject.class, "p");
+        List<Class> scmTypes = new ArrayList<Class>();
+        List<SCMDescriptor<?>> scmDescriptors = SingleSCMSource.DescriptorImpl.getSCMDescriptors(mp);
+        for (SCMDescriptor<?> scmDescriptor : scmDescriptors) {
+            scmTypes.add(scmDescriptor.clazz);
+        }
+        assertThat(scmTypes, hasItem((Class) GitSCM.class));
+        assertThat(scmTypes, not(hasItem((Class) OldSCM.class)));
+        /* More realistic variant:
```

Comment:
Well my initial attempt at a fix was to use the equivalent of `@AncestorInPath` to look up the `WorkflowJob` instance. For that, a more realistic test was needed. Then I realized that such a trick would not work anyway, since when configuring a new multibranch project there is no branch project yet. So I just kept this in case some future iteration of the fix actually is more sensitive to the request environment.